keps/draft-20190418-skeleton-generator.md

```diff
@@ -0,0 +1,39 @@
+---
+kep-number: draft-20190418
+title: Skeleton generator
+authors:
+  - "@djannot"
+owners:
+  - TBD
+editor: TBD
+creation-date: 2019-04-18
+status: provisional
+---
+
+# Skeleton generator
+
+## Table of Contents
+
+  * [Table of Contents](#table-of-contents)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+
+Created by [gh-md-toc](https://github.com/ekalinin/github-markdown-toc)
+
+## Summary
+
+In order to develop a new framework using KUDO, you always need to create several `Plans` (`deploy`, `update`, `upgrade`, ...) and for each plan to create several `Phases`, `Steps` and `Tasks`.
+
+Also, any Framework will have a `Task` in the `deploy` plan to create similar Kubernetes objects (`StatefulSets`, `Services`, ...).
+
+A Skeleton generator could create `yaml` templates with the common `Plans`, `Phases`, `Steps` and `Tasks`.
+
+This generator should be part of the KUDO CLI.
```

Comment:
agree and think of a `kubectl kudo init` command that might even help boilerplate entire repositories.